demos/python_demos/human_pose_estimation_3d_demo/human_pose_estimation_3d_demo.py

```diff
@@ -0,0 +1,145 @@
+                      help='Show this help message and exit.')
+    args.add_argument('-m', '--model',
+                      help='Required. Path to an .xml file with a trained model '
+                           '(or .pth in case of PyTorch inference).',
+                      type=str, required=True, metavar='"<path>"')
+    args.add_argument('--video', help='Optional. Path to video file or camera id.', type=str, default='')
+    args.add_argument('-d', '--device',
+                      help='Optional. Specify the target device to infer on: CPU, GPU, FPGA, HDDL or MYRIAD. '
+                           'The demo will look for a suitable plugin for device specified '
+                           '(by default, it is CPU).',
+                      type=str, default='CPU', metavar='"<path>"')
+    args.add_argument('--use-pytorch',
+                      help='Optional. Run network with PyTorch as inference engine. '
+                           'Only CPU and GPU devices are supported.',
+                      action='store_true')
+    args.add_argument('--images', help='Optional. Path to input image(s).', nargs='+', default='')
+    args.add_argument('--height-size', help='Optional. Network input layer height size.', type=int, default=256)
+    args.add_argument('--extrinsics-path',
+                      help='Optional. Path to file with camera exstrinsics.',
+                      type=str, default=None, metavar='"<path>"')
+    args.add_argument('--fx', type=np.float32, default=-1, help='Optional. Camera focal length.')
+    args = parser.parse_args()
+
+    if args.video == '' and args.images == '':
+        raise ValueError('Either --video or --image has to be provided')
+
+    stride = 8
+    if not args.use_pytorch:
+        net = InferenceEngineOpenVINO(args.model, args.device, stride)
+    else:
+        from modules.inference_engine_pytorch import InferenceEnginePyTorch
+        net = InferenceEnginePyTorch(args.model, args.device, stride)
+
+    canvas_3d = np.zeros((720, 1280, 3), dtype=np.uint8)
+    plotter = Plotter3d(canvas_3d.shape[:2])
+    canvas_3d_window_name = 'Canvas 3d'
+    cv2.namedWindow(canvas_3d_window_name)
+    cv2.setMouseCallback(canvas_3d_window_name, Plotter3d.mouse_callback)
+
+    file_path = args.extrinsics_path
+    if file_path is None:
+        file_path = os.path.join('data', 'extrinsics.json')
+    with open(file_path, 'r') as f:
+        extrinsics = json.load(f)
+    R = np.array(extrinsics['R'], dtype=np.float32)
+    t = np.array(extrinsics['t'], dtype=np.float32)
+
+    frame_provider = ImageReader(args.images)
+    is_video = False
+    if args.video != '':
+        frame_provider = VideoReader(args.video)
+        is_video = True
+    base_height = args.height_size
+    fx = args.fx
+
+    delay = 1
+    esc_code = 27
+    p_code = 112
+    mean_time = 0
+    for frame in frame_provider:
+        current_time = cv2.getTickCount()
+        if frame is None:
+            break
```

Comment:
It does not seem that that `break` is normally reached. `__next__` will throw an exception instead of providing an empty frame.